lib/index.d.ts

```diff
@@ -296,6 +296,14 @@ declare namespace Realm.Sync {
         static registerWithProvider(server: string, options: { provider: string, providerToken: string, userInfo: any }, callback: (error: Error | null, user: User | null) => void): void;
         static registerWithProvider(server: string, options: { provider: string, providerToken: string, userInfo: any }): Promise<Realm.Sync.User>;
 
+        static requestPasswordReset(server: string, email: string): Promise<Void>;
+
+        static completePasswordReset(server:string, reset_token:string, new_passwd:string): Promise<Void>;
```

Comment:
Why `new_passwd`? It's not hard to guess what it means, but still, don't see much value in saving two characters. Also, it's inconsistent with other methods that have `password` spelled out correctly.